src/components/ReportsPage/HeatmapPage/HeatmapTable.tsx

```diff
@@ -0,0 +1,226 @@
+ * limitations under the License.
+ */
+import React, { useCallback, useMemo } from 'react';
+import { EmptyState } from '@backstage/core-components';
+import {
+  useCortexBirdseye,
+  BirdsEyeReportTable,
+  TeamResponse,
+  Domain,
+  HierarchyResponse,
+  StringIndexable,
+  TeamDetails,
+  Filters,
+  BirdsEyeReportTableProps,
+} from '@cortexapps/birdseye';
+import {
+  DomainHierarchiesResponse,
+  Scorecard,
+  ScorecardLadder,
+  ScorecardServiceScore,
+} from '../../../api/types';
+import { HomepageEntity } from '../../../api/userInsightTypes';
+import { mapKeys, mapValues } from 'lodash';
+import { HeatmapSettings } from './HeatmapSettings';
+import { useColorCellStyles } from './colorClasses';
+import { Grid } from '@material-ui/core';
+import { useRouteRef } from '@backstage/core-plugin-api';
+import { scorecardServiceDetailsRouteRef } from '../../../routes';
+import {
+  defaultComponentRefContext,
+  entityComponentRef,
+} from '../../../utils/ComponentUtils';
+import { parseEntityRef } from '@backstage/catalog-model';
+import {
+  BirdsEyeAnchorAdapter,
+  convertToBirdsEyeDomainHierarchy,
+  convertToBirdsEyeScorecard,
+  convertToBirdsEyeScores,
+  getCellColorBackground,
+  getScoreColorClassName,
+} from './HeatmapUtils';
+
+interface HeatmapTableProps {
+  allDomains: Domain[];
+  allTeams: TeamResponse[];
+  catalog: HomepageEntity[];
+  domainHierarchy: DomainHierarchiesResponse | undefined;
+  domainAncestryMap: Record<number, number[]>;
+  entitiesByTag: StringIndexable<HomepageEntity>;
+  filters: Filters;
+  ladder?: ScorecardLadder;
+  scorecard: Scorecard;
+  scores: ScorecardServiceScore[];
+  setFilters: (filters: Filters) => void;
+  teamHierarchy?: HierarchyResponse;
+  teamsByEntity: StringIndexable<TeamDetails[]> | undefined;
+}
+
+export const HeatmapTable: React.FC<HeatmapTableProps> = ({
+  allDomains,
+  allTeams,
+  catalog,
+  domainAncestryMap,
+  domainHierarchy,
+  entitiesByTag,
+  filters,
+  ladder,
+  scorecard,
+  scores,
+  setFilters,
+  teamsByEntity,
+  teamHierarchy,
+}) => {
+  const colorStyles = useColorCellStyles();
+
+  const {
+    mappedDomainAncestryMap,
+    mappedDomainHierarchies,
+    mappedScorecard,
+    mappedScores,
+  } = useMemo(() => {
+    const mappedScorecard = convertToBirdsEyeScorecard(scorecard, ladder);
+    const mappedScores = convertToBirdsEyeScores(scores, catalog);
+    const mappedDomainHierarchies =
+      convertToBirdsEyeDomainHierarchy(domainHierarchy);
+    const mappedDomainAncestryMapKeys = mapKeys(domainAncestryMap, key =>
+      key.toString(),
+    );
+    const mappedDomainAncestryMap = mapValues(
+      mappedDomainAncestryMapKeys,
+      values => values.map(value => value.toString()),
+    );
+
+    return {
+      mappedScorecard,
+      mappedScores,
+      mappedDomainHierarchies,
+      mappedDomainAncestryMap,
+    };
+  }, [catalog, domainAncestryMap, domainHierarchy, ladder, scorecard, scores]);
```

Comment:
Not sure why we are doing some of these things? The birdseye report should take care of it all?
Can we just adjust the backstage endpoints to a v2 and have it mapped directly instead of having to do a bunch of conversion

Reply:
There are some differences how some types are defined in the Backstage plugin and in the Cortex app (and the Bird's eye package). The biggest mess makes the compound filter definition. The type is defined as a string in the Cortex app, but in the backstage plugin is an enum.
```
// Backstage plugin
interface CompoundFilter {
  cqlFilter?: GenericCqlFilter;
  entityGroupFilter?: EntityGroupFilter;
  type: FilterType.COMPOUND_FILTER;              // << HERE
  typeFilter?: CatalogPageTypeFilter;
}

// Cortex app
interface CompoundFilter {
  cqlFilter?: GenericCqlFilter;
  entityGroupFilter?: EntityGroupFilter;
  type: "COMPOUND_FILTER";                       // << HERE
  typeFilter?: CatalogPageTypeFilter | null;
}
```
And when I tried to convert it, it caused a lot of consequences and I'm not sure if I should update it within this PR. And then also domain tree has a different object shape in the plugin and also some properties are optional or nullable in one application and not in the other one.

Reply:
yeah this is just an enum issue, so it's fine.